src/redisraft.h

```diff
@@ -558,8 +561,18 @@ typedef struct ShardingInfo {
     ShardGroup *stable_slots_map[REDIS_RAFT_HASH_SLOTS];
     ShardGroup *importing_slots_map[REDIS_RAFT_HASH_SLOTS];
     ShardGroup *migrating_slots_map[REDIS_RAFT_HASH_SLOTS];
+
+    raft_term_t max_importing_term[REDIS_RAFT_HASH_SLOTS];
 } ShardingInfo;
 
+typedef struct {
+    raft_term_t term;
+    int magic;
```

Comment:
I don't remember what will we use for `magic` number but we should be careful not to cause overflow (undefined behavior). I suggest using unsigned numbers for this purpose or even fixed length integers.

Reply:
magic was just a shared value for the importing/migrating shardgroup to let importing know that its getting a request from a migrating one that is on the same "round" as it.  (not from something stale for some reason).

Reply:
Alternatively, we can declare `magic` as `0 <= magic <= 2^31` , still need to do check before casting:

```c
    if (magic < 0 || magic > INT32_MAX) {
        ReplyError();
    }
```

Reply:
I just made magic take a long long to fit with redis module api.